include/odp/api/spec/std_types.h

```diff
@@ -40,6 +41,25 @@ extern "C" {
  */
 typedef uint32_t odp_percent_t;
 
+/**
+ * 128-bit unsigned integer structure.
+ */
+struct odp_u128_s{
+	union {
+		uint64_t u64[2];
+		uint32_t u32[4];
+		uint16_t u16[8];
+		uint8_t  u8[16];
+	};
+} ODP_ALIGNED(16);
+
+/**
+ * @typedef odp_u128_t
+ * 128-bit unsigned integer
+ */
+typedef struct odp_u128_s odp_u128_t;
```

Comment:
I'd prefer typedef combined with the struct definition above. Also align attribute should follow "struct" keyword. u64/u32/... variables need to be doxygen documented.

```
/**
 * 128-bit unsigned integer
*/
typedef struct ODP_ALIGNED(16) odp_u128_t{
	union {
		/** 128 bits as uint64_t words */
		uint64_t u64[2];
		/** 128 bits as uint32_t words */
		uint32_t u32[4];
		/** 128 bits as uint16_t words */
		uint16_t u16[8];
		/** 128 bits as bytes */
		uint8_t  u8[16];
	};
} odp_u128_t;
```

Reply:
I made the suggested changes. Do I need to follow this convention only in spec files ?
I took the reference from "include/odp/api/abi-default/atomic.h":
struct odp_atomic_u64_s {
        uint64_t v; /**< Actual storage for the atomic variable */
        /* Some architectures do not support lock-free operations on 64-bit
         * data types. We use a spin lock to ensure atomicity. */
        char lock; /**< Spin lock (if needed) used to ensure atomic access */
} ODP_ALIGNED(sizeof(uint64_t)); /* Enforce alignment! */

typedef struct odp_atomic_u64_s odp_atomic_u64_t;

Reply:
Most our typedef structs combine typedef and struct definition. So, I'd do it the same way also in ABI or implementation files, if these no reason to separate those (forward type definition).

Align attribute after struct keyword is from GCC manual, so that order should be always applied. Atomic ABI file does not currently but may be you can fix it while at it.

Reply:
Sure. We will do similar changes in implementation files as a separate commit.